test/E2E/Azure.Functions.PowerShellWorker.E2E/Azure.Functions.PowerShellWorker.E2E/Utilities.cs

```diff
@@ -53,5 +47,27 @@ public static async Task<bool> InvokeHttpTrigger(string functionName, string que
             }
             return true;
         }
+
+        public static async Task<string> InvokeHttpTrigger(string functionName, string queryString, HttpStatusCode expectedStatusCode, int expectedCode = 0)
+        {
+            var response = await GetHttpTriggerResponse(functionName, queryString);
+            if (expectedStatusCode != response.StatusCode && expectedCode != (int)response.StatusCode)
+            {
+                return null;
+            }
+
+            return await response.Content.ReadAsStringAsync();
+        }
+
+        private static async Task<HttpResponseMessage> GetHttpTriggerResponse(string functionName, string queryString)
+        {
+            string uri = $"api/{functionName}{queryString}";
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, uri);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("text/plain"));
+
+            var httpClient = new HttpClient();
```

Comment:
Pretty sure we could reuse an HttpClient that way we’re not allocating one per-request

Reply:
We definitely could, but I'm not convinced the benefits (saving a few milliseconds per test run?) outweigh the drawbacks (increased coupling between tests). Unless you find a really good reason, I will leave it as is.

Reply:
That's fine, just wanted to point it out.

Reply:
Please consider using the ```using statement``` for the httpclient. Other than that, LGTM.

Reply:
> Please consider using the `using statement` for the httpclient. Other than that, LGTM.

Thank you, let's do it next time. This PR does not make anything worse: this is not new code, I'm just extracting a method.

Reply:
Sounds good. Thanks @AnatoliB